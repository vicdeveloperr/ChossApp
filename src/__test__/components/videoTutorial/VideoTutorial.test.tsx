import {
  VideoTutorial,
  videoRef,
} from "../../../components/videoTutorial/VideoTutorial";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react-native";

describe("<VideoTutorial />", () => {
  it("Renderiza el componente VideoTutorialPlayer con la estructura esperada", () => {
    const tree = renderer.create(<VideoTutorial sourceUri="" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("Define valor de videoRef correctamente", () => {
    render(<VideoTutorial sourceUri="" />);

    expect(videoRef).toBeDefined();
  });
});
