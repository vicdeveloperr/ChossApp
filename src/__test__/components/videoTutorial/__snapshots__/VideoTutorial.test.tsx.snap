// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoTutorial /> Renderiza el componente VideoTutorialPlayer con la estructura esperada 1`] = `
<View
  pointerEvents="box-none"
  style={
    [
      {
        "overflow": "hidden",
      },
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
    ]
  }
>
  <ViewManagerAdapter_ExpoVideoView
    onError={[Function]}
    onFullscreenUpdate={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onReadyForDisplay={[Function]}
    onStatusUpdate={[Function]}
    resizeMode="mock"
    source={
      {
        "headers": undefined,
        "overridingExtension": null,
        "uri": "",
      }
    }
    status={
      {
        "isLooping": true,
        "shouldPlay": true,
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
    videoStyle={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`Renderiza el componente VideoTutorialPlayer con la estructura esperada 1`] = `
<View
  pointerEvents="box-none"
  style={
    [
      {
        "overflow": "hidden",
      },
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
    ]
  }
>
  <ViewManagerAdapter_ExpoVideoView
    onError={[Function]}
    onFullscreenUpdate={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onReadyForDisplay={[Function]}
    onStatusUpdate={[Function]}
    resizeMode="mock"
    source={
      {
        "headers": undefined,
        "overridingExtension": null,
        "uri": "",
      }
    }
    status={
      {
        "isLooping": true,
        "shouldPlay": true,
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
    videoStyle={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  />
</View>
`;
